### ISSUE TITLE
PDB++ arrow/tab keys don't work with allennlp

### ISSUE THREAD

**Issue by successar** [allenai/allennlp#4378]
## Checklist

- [x] I have verified that the issue exists against the `master` branch of AllenNLP.
- [ ] I have read the relevant section in the [contribution guide](https://github.com/allenai/allennlp/blob/master/CONTRIBUTING.md#bug-fixes-and-new-features) on reporting bugs.
- [x] I have checked the [issues list](https://github.com/allenai/allennlp/issues) for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/allenai/allennlp/pulls) for existing proposed fixes.
- [x] I have checked the [CHANGELOG](https://github.com/allenai/allennlp/blob/master/CHANGELOG.md) and the [commit log](https://github.com/allenai/allennlp/commits/master) to find out if the bug was already fixed in the master branch.
- [x] I have included in the "Description" section below a traceback from any exceptions related to this bug.
- [x] I have included in the "Related issues or possible duplicates" section beloew all related issues and possible duplicate issues (If there are none, check this box anyway).
- [x] I have included in the "Environment" section below the name of the operating system and Python version that I was using when I discovered this bug.
- [x] I have included in the "Environment" section below the output of `pip freeze`.
- [x] I have included in the "Steps to reproduce" section below a minimally reproducible example.

## Description

I can't use pdb/pdb++ arrow keys or tab key for autocompletion when running `allennlp train` . Possibly related to this issue https://stackoverflow.com/a/56418204

</details>

## Related issues or possible duplicates

- Reopening this issue - https://github.com/allenai/allennlp/issues/2176

## Environment

OS: Linux Ubuntu 18.04

Python version: 3.7.7

<details>
<summary><b>Output of <code>pip freeze</code>:</b></summary>
<p>

```
allennlp==1.0.0
appdirs==1.4.3
astroid==2.4.1
attrs==19.3.0
black==19.10b0
blis==0.4.1
boto3==1.14.4
botocore==1.17.4
catalogue==1.0.0
certifi==2020.4.5.2
chardet==3.0.4
click==7.1.2
cymem==2.0.3
docutils==0.15.2
fancycompleter==0.9.1
filelock==3.0.12
future==0.18.2
gnureadline==8.0.0
h5py==2.10.0
idna==2.9
importlib-metadata==1.6.1
isort==4.3.21
jmespath==0.10.0
joblib==0.15.1
jsonnet==0.16.0
jsonpickle==1.4.1
lazy-object-proxy==1.4.3
mccabe==0.6.1
more-itertools==8.4.0
murmurhash==1.0.2
mypy-extensions==0.4.3
nltk==3.5
numpy==1.18.5
overrides==3.0.0
packaging==20.4
pathspec==0.7.0
pdbpp==0.10.2
plac==1.1.3
pluggy==0.13.1
preshed==3.0.2
protobuf==3.12.2
py==1.8.2
Pygments==2.6.1
pylint==2.5.2
pyparsing==2.4.7
pyrepl==0.9.0
pytest==5.4.3
python-dateutil==2.8.1
regex==2020.6.8
requests==2.23.0
s3transfer==0.3.3
sacremoses==0.0.43
scikit-learn==0.23.1
scipy==1.4.1
sentencepiece==0.1.91
six==1.15.0
spacy==2.2.4
srsly==1.0.2
tensorboardX==2.0
thinc==7.4.0
threadpoolctl==2.1.0
tokenizers==0.7.0
toml==0.10.0
torch==1.5.0
tqdm==4.46.1
transformers==2.11.0
typed-ast==1.4.1
typing-extensions==3.7.4.1
urllib3==1.25.9
wasabi==0.6.0
wcwidth==0.2.4
wmctrl==0.3
wrapt==1.11.2
zipp==3.1.0
```

</p>
</details>

## Steps to reproduce

1. Install allennlp in clean environment
2. Run a short script and try arrow keys

```python
v = 10
breakpoint()
```

Arrow keys should work

3. Next run a `allennlp train` script and again put breakpoint somewhere. Check arrow keys.

<details>
<summary><b>Example source:</b></summary>
<p>

```
```

</p>
</details>

**Comment by dirkgr** [comment]
I put a workaround here: https://github.com/allenai/allennlp/pull/4383

With that PR, if you run with the `--no-redirect-std` flag, it won't do the redirection and arrow keys will work as expected. I'm trying to find out if we can just always run that way instead of having a new flag.

**Comment by dirkgr** [comment]
This is now fixed as of #4383.
